metricbeat/beater/reload.go

```diff
@@ -0,0 +1,252 @@
+package beater
```

Comment:
I wonder if the code in this file could be generalized into a libbeat package. It seems to me that it contains relatively little Metricbeat specific logic, except for the `StartModules` and `StopModules` functions which can be easily injected.

I'm thinking that we'll want to make it easy to extend this type of reloading functionality to other beats.